src/mqtt.c

```diff
@@ -238,6 +238,20 @@ static void on_message(
   sfree(vl.values);
 } /* void on_message */
 
+static int mqtt_subscribe(mqtt_client_conf_t *conf) {
+  int status;
+
+  status = mosquitto_subscribe(conf->mosq, /* message_id = */ NULL, conf->topic,
+                               conf->qos);
+  if (status != MOSQ_ERR_SUCCESS) {
+    ERROR("mqtt plugin: Subscribing to \"%s\" failed: %s", conf->topic,
+          mosquitto_strerror(status));
+    return -1;
+  }
```

Comment:
Both call sites call `mosquitto_disconnect(conf->mosq);` on failure. Should this be deduplicated here, too?

Reply:
Okay.